tests/test_distributed/test_with_spark/test_spark_local.py

```diff
@@ -1630,6 +1630,34 @@ def test_unsupported_params(self):
         with pytest.raises(ValueError, match="evals_result"):
             SparkXGBClassifier(evals_result={})
 
+    def test_tracker(self):
+        classifier = SparkXGBClassifier(
+            launch_tracker_on_driver=True,
+            tracker_host="192.168.1.32",
+            tracker_port=59981,
+        )
+        with pytest.raises(Exception, match="Failed to bind socket"):
+            classifier._get_tracker_args()
+
+        classifier = SparkXGBClassifier(
+            launch_tracker_on_driver=False, tracker_host="127.0.0.1", tracker_port=58892
+        )
+        with pytest.raises(
+            ValueError, match="You must enable launch_tracker_on_driver"
```

Comment:
Why does it need to be on the driver in order to have a custom host?

Reply:
We all know that the tracker is launched on the task 0 on the executor, but the issue is that we actually have no idea which worker the task 0 will be scheduled into. So naturally,  we don't know the IP address for task 0.  Then setting tracker host will not work for this case. But if the tracker is launched on the host, then we can set the tracker host and port.